packages/core/utils/src/auth/abstract-auth-provider.ts

```diff
@@ -26,11 +106,195 @@ export abstract class AbstractAuthModuleProvider implements IAuthProvider {
       config.displayName
   }
 
+  /**
+   * This method authenticates the user.
+   * 
+   * If the authentication requires an action to perform with a third-party service, such as login with
```

Comment:
"The authentication happens either by directly authenticating or returning a redirect URL to continue the authentication with a third party provider."

Or something along those lines, to make it a bit clearer that there are basically 2 ways `authenticate` behaves